CDP4WebServices.API/Services/ChangeLog/ChangeLogService.cs

```diff
@@ -0,0 +1,1311 @@
+        /// The injected <see cref="ICdp4TransactionManager"/>
+        /// </summary>
+        public ICdp4TransactionManager TransactionManager { get; set; }
+
+        /// <summary>
+        /// Appends changelog data based on the changed <see cref="CDP4Common.DTO.Thing"/>s
+        /// </summary>
+        /// <param name="transaction">
+        /// The current <see cref="NpgsqlTransaction"/> to the database.
+        /// </param>
+        /// <param name="partition">
+        /// The database partition (schema) where the requested resource is stored.
+        /// </param>
+        /// <param name="actor">
+        /// The <see cref="Person.Iid"/> of the person that made the changes.
+        /// </param>
+        /// <param name="transactionRevision">
+        /// The revisionNumber of the <see cref="transaction"/>
+        /// </param>
+        /// <param name="operation">
+        /// <see cref="CdpPostOperation"/> that resulted to all the changes.
+        /// </param>
+        /// <param name="things">
+        /// The <see cref="IReadOnlyList{T}"/> of type <see cref="CDP4Common.DTO.Thing"/> that contains changed <see cref="CDP4Common.DTO.Thing"/>
+        /// </param>
+        /// <returns>
+        /// True if change log data was added, otherwise false
+        /// </returns>
+        public bool TryAppendModelChangeLogData(NpgsqlTransaction transaction, string partition, Guid actor, int transactionRevision, CdpPostOperation operation, IReadOnlyList<Thing> things)
+        {
+            var isFullAccessEnabled = this.TransactionManager.IsFullAccessEnabled();
+            var result = false;
+
+            if (!isFullAccessEnabled)
+            {
+                this.TransactionManager.SetFullAccessState(true);
+            }
+
+            try
+            {
+                var changedEngineeringModels =
+                    things
+                        .Where(x => x.ClassKind == ClassKind.EngineeringModel)
+                        .Cast<EngineeringModel>()
+                        .ToList();
+
+                if (changedEngineeringModels.Count == 1)
+                {
+                    var engineeringModel = changedEngineeringModels.First();
+                    var changedThings = things.Where(x => x.RevisionNumber == transactionRevision).ToList();
+
+                    var modelLogEntries =
+                        operation
+                            .Create
+                            .Where(x => x.ClassKind == ClassKind.ModelLogEntry)
+                            .Cast<ModelLogEntry>()
+                            .ToList();
+
+                    ModelLogEntry modelLogEntry;
+
+                    if (modelLogEntries.Count != 1)
+                    {
+                        var modelLogOperationData = new CdpPostOperation();
+
+                        var newModelLogEntry = new ModelLogEntry(Guid.NewGuid(), 0)
+                        {
+                            Level = LogLevelKind.USER,
```

Comment:
as per 10-25 https://sdk-reference.cdp4.org/#LogLevelKind i think this should not be USER,. i propose INFO